Year and month modal display on navigation to SSD

dborowiecki opened the pull request.

orangejenny committed:
Updated signature of build_application_zip

orangejenny committed:
Removed overly broad exception handling from app building

orangejenny committed:
Added named variables to translation strings

Commit:
implement service document for case OData from configs

Commit:
implement metadata document for case OData from configs

Commit:
implement empty feed

Commit:
not tested but required

Commit:
serialize case data

Commit:
implement @odata.nextLink

Commit:
filter by case type [no unit tests]

Commit:
get OData link from UI

Rohit25negi committed:
Add THR report

Rohit25negi committed:
Add thr report generator in task

Rohit25negi committed:
Accept request for THR report

Rohit25negi committed:
Corrected Logic

Rohit25negi committed:
Added test case

Rohit25negi committed:
Fix lynter

Rohit25negi committed:
Better indentation

Commit:
[ICDS] month filter in SSD report start from Feb 2018

Commit:
linter

Commit:
demo sync test failures in python 3

Commit:
use OrderedDict so tests in python 2/3 produce identical output

Commit:
fix tests by setting CaseExportInstance.case_type so ODataCaseFromExportInstanceResource.obj_get_list succeeds

mkangia committed:
make practice_mobile_worker visible on settings page for Linked App

mkangia committed:
add tests to confirm not overridden on pull/update

Rohit25negi committed:
Merge branch 'rn_thr_report_view' into rn_generate_thr_report

Rohit25negi committed:
Use django ORM

Rohit25negi committed:
Fix lynter

biyeun committed:
update text on feature previews page

mkangia committed:
create targeted ccz if build has commcare flavor set

Rohit25negi committed:
CLean code

emord committed:
Fix encoder

emord committed:
Fix data source engine override

emord committed:
Update userreports migrations

sravfeyn committed:
Don't commit without checking

sravfeyn committed:
add option to specify custom pillow reorg

millerdev committed:
Disambiguate parent/child relationship

proteusvacuum committed:
Remove memoization on fields that can change

proteusvacuum committed:
Clear reverse indices memoization when necessary

proteusvacuum committed:
Memoize other index tree methods

esoergel committed:
Make conditionally_location_safe importer appear

https://dimagi-dev.atlassian.net/browse/PF-389
This makes the link to the case importer appear for location-restricted
users on domains with the feature flag LOCATION_SAFE_CASE_IMPORTS
enabled.

When uitabs checks if a URL is location safe, it does not pass `request`
to the conditional function (since request params will vary).  This only needed
access to domain, which is already available in `kwargs`, so a switch to
that makes it work.

Commit:
reference .in files inside of other .in files

Commit:
make requirements for relevant python version

Rohit25negi committed:
Remove extra argument

Rohit25negi committed:
Merge branch 'master' into rn_generate_thr_report

Commit:
[ICDS] Fixed showing message when going through between reports

mkangia committed:
Merge pull request #24706 from dimagi/sr-pillow-reset

Ability to specify custom pillow reorg mapping and run check mode by default

millerdev committed:
Merge pull request #24707 from dimagi/dm/child-comment

Disambiguate parent/child relationship

orangejenny committed:
Updated error messages in test to match previous commit

nickpell committed:
Merge pull request #24712 from dimagi/make-requirements

pip-compile should update the requirements.txt matching the Python version

snopoke committed:
fix comparison of data with Decimals and dates

esoergel committed:
Merge pull request #24711 from dimagi/es/conditionally_safe

Make conditionally_location_safe importer appear

Commit:
cleanup correct directory

Commit:
don't block indefinitely during tests if there is a long diff

orangejenny committed:
Merge pull request #24617 from dimagi/jls/fix-translations

Added named variables to translation strings

Commit:
run tests in Python 3 if flag is provided in test matrix

emord committed:
Merge pull request #24702 from dimagi/je/fix-encoder

Fix encoder for citus/UCR comparisons

orangejenny committed:
Update staging.yaml

orangejenny committed:
Update staging.yaml

mkangia committed:
merge master

mkangia committed:
remove duplicate definition after resolving conflict

orangejenny committed:
Update staging.yaml

esoergel committed:
Coerce falsy values to null

https://dimagi-dev.atlassian.net/browse/HI-716
Fixes an issue where changing module settings would trigger a reorder
This happened only with a particular combination of feature flags and
addons.  In particular, BASIC_CHILD_MODULE had to be disabled.  At any
rate, the issue was that `oldRoot` was `""` and `newRoot` was
`undefined`, since the the select parent widget was only available for
advanced modules.  Since `"" !=== undefined`, this incorrectly triggered
a reorder.

biyeun committed:
handle loss of internet connection when submitting forms in webapps

orangejenny committed:
Updated test value for flavor to be more realistic

biyeun committed:
elaborate on error message

mkangia committed:
better naming

mkangia committed:
Merge pull request #24691 from dimagi/mk/allow-practice-worker-on-la

Allow setting practice mobile worker on Linked App

esoergel committed:
exit early if parent widget isn't present

biyeun committed:
modify offline check

biyeun committed:
add additional UX that disables the form completely when the window is offline and shows an error and then enables the form and shows a success message when back online

orangejenny committed:
Removed advanced modules warning

orangejenny committed:
Added except blocks for CaseError thrown when getting case properties

nickpell committed:
Merge pull request #24714 from dimagi/fix-test-make-requirements

Fix test-make-requirements.sh

orangejenny committed:
Merge pull request #24718 from dimagi/jls/tweak-flavor-in-test

Updated test value for flavor to be more realistic

Commit:
[refactor] use TableConfiguration.selected_columns

orangejenny committed:
Update staging.yaml

esoergel committed:
Codify CCHQ_IS_FRESH_INSTALL for wider use

Commit:
render feed using same code as exports

nickpell committed:
Merge pull request #24722 from dimagi/test-flag

run tests in Python 3 if flag is provided in test matrix

Commit:
404 if url in domain does not match config's domain

Commit:
always setup ES index and user permission

Commit:
missing config id should result in 404

Commit:
test correct condition in test_config_in_different_domain

Commit:
test user permissions and fix them for service and metadata views

Rohit25negi committed:
Merge pull request #24654 from dimagi/rn_generate_thr_report

Generate thr excel report(xlsx/csv)

mkangia committed:
Merge pull request #24719 from dimagi/mk/better-naming

better naming

snopoke committed:
sql-agg with simple columns called without group_by

snopoke committed:
Missing form transaction during rebuild

snopoke committed:
form edit assert

Commit:
replace starting date from Feb 2018 to Feb 2019 in the month/year filter in SDD

Rohit25negi committed:
Use Person case for dob

Rohit25negi committed:
Correct the fixture

Rohit25negi committed:
Fix test cases

orangejenny committed:
Merge pull request #24721 from dimagi/jls/catch-case-error

Added except blocks for CaseError thrown when getting case properties

orangejenny committed:
Merge pull request #24720 from dimagi/jls/remove-warning

Removed advanced modules warning

orangejenny committed:
Update staging.yaml

mkangia committed:
Add note field

mkangia committed:
update toggle description for easy search of ff

biyeun committed:
Merge pull request #24716 from dimagi/bmb/webapps-offline

handle loss of internet connection when submitting forms in webapps

Commit:
stickler

orangejenny committed:
Added parent module, if present, to module's frame children

proteusvacuum committed:
Revert "demo sync test failures in python 3"

This reverts commit cd42de747ffbcf67c08a3eb6a427ba5d50f64ffe.

orangejenny committed:
Added parent/child menu test

Commit:
refactor class names for brevity

biyeun committed:
make column title optional. if not filled out, default to property name

biyeun committed:
only show table loading text after initial load has finished

biyeun committed:
fix typo

esoergel committed:
Merge pull request #24725 from dimagi/es/fresh-installs

Codify CCHQ_IS_FRESH_INSTALL for wider use

Commit:
remove obsolete comment

nickpell committed:
Merge pull request #24731 from dimagi/remove-outdated-comment

remove obsolete comment

mkangia committed:
fix tests

mkangia committed:
lint

biyeun committed:
make sure that the table's minimum height is always 200px

biyeun committed:
make sure that on initial load we obey the goToPage rule

mkangia committed:
just use blank check

biyeun committed:
handle es timeouts more gracefully

biyeun committed:
handle negative numbers in numeric comparison

biyeun committed:
update learn more link

orangejenny committed:
Removed J2ME app settings from UI for non-J2ME builds

biyeun committed:
Merge pull request #24695 from dimagi/bmb/feature-previews-update

update text on feature previews page

orangejenny committed:
Updated test data

proteusvacuum committed:
Allow negative numbers in comparisons and equality operations

proteusvacuum committed:
Merge pull request #24709 from dimagi/fr/cousins-sync-bug

Cousins sync bug

nickpell committed:
Merge pull request #24686 from dimagi/odata-merge

configure case OData feed

Commit:
send OData metrics to datadog

Commit:
allow setting OData feature flag at user level

esoergel committed:
Merge pull request #24715 from dimagi/es/module-reorder

Don't trigger a reorder when changing module settings

Commit:
update staging.yaml

[ci skip]

Commit:
increase max limit of case OData feed

Commit:
default OData feed limit to 2000

Commit:
[refactor] get odataUrl in separate function

orangejenny committed:
Merge pull request #24730 from dimagi/jls/eof-nav-with-parent

Update end of form nav "previous screen" to support parent/child menus

orangejenny committed:
Merge pull request #24734 from dimagi/jls/disable-autofill

Removed J2ME app settings from UI for non-J2ME builds

nickpell committed:
Merge pull request #24738 from dimagi/odata-user

allow setting OData feature flag at user level

mkangia committed:
better naming and skip comment in favour of followup PR

mkangia committed:
Merge pull request #24728 from dimagi/mk/add-note-field

Add note field

mkangia committed:
Merge pull request #24708 from dimagi/mk/add-flavour

create targeted ccz if build has commcare flavor set

orangejenny committed:
Update staging.yaml

snopoke committed:
allow sorting by indexed_on

orangejenny committed:
Added .ko-template to manage lookup tables page

orangejenny committed:
Removed ajax call to fetch tables

orangejenny committed:
Removed flag

orangejenny committed:
Updated text to be more specific

orangejenny committed:
Removed title casing

It was turning "SMS" into "Sms".

orangejenny committed:
Merge pull request #24473 from dimagi/jls/remove-overbroad-catch-redux

Removed overly broad exception handling from app building

nickpell committed:
Merge pull request #24739 from dimagi/odata-limit

set custom max_limit and default limit for OData case feed

orangejenny committed:
Added flaky

nickpell committed:
Merge pull request #24737 from dimagi/odata-datadog

send OData metrics to datadog

orangejenny committed:
Updated page name

snopoke committed:
Merge pull request #24713 from dimagi/sk/labratory-json

fix comparison of data with Decimals and dates

snopoke committed:
Merge pull request #24726 from dimagi/sk/soft-asserts

remove soft asserts

czue committed:
remove hope api

orangejenny committed:
Updated help link

orangejenny committed:
Added to dev-requirements

orangejenny committed:
Moved decorator to encompass whole class

orangejenny committed:
Merge pull request #24742 from dimagi/jls/fixtures-ui-tweaks

Manage lookup tables page: stop fetching tables twice

czue committed:
update expired domains

biyeun committed:
Merge pull request #24736 from dimagi/fr/negative-numbers-case-list-explorer-dsl

Allow negative numbers in comparisons and equality operations

orangejenny committed:
Merge pull request #24743 from dimagi/jls/unflag-bulk-sms

Removed conditional alerts flag

orangejenny committed:
Added download support

emord committed:
Delete RHEL-INSTALL.rst

emord committed:
Merge pull request #24745 from dimagi/cz/kill-hope-api

remove hope api

emord committed:
Merge pull request #24746 from dimagi/emord-patch-1-1

Delete RHEL-INSTALL.rst

orangejenny committed:
Added upload support

biyeun committed:
Merge pull request #24732 from dimagi/bmb/ecd-updates

Fix remaining bugs & requests in Explore Case Data report

orangejenny committed:
Added download test

orangejenny committed:
Added upload test

Commit:
update commcare-translations

emord committed:
Remove missing doc task

Commit:
update staging.yaml

[ci skip]

nickpell committed:
Merge pull request #24748 from dimagi/update-commcare-translations

update commcare-translations

mkangia committed:
Add branch for staging

snopoke committed:
change submission_type tag from 'unknown' to 'error'

sravfeyn committed:
Fix the method to return a boolean

snopoke committed:
Merge pull request #24741 from dimagi/sk/indexed_on_sort

allow sorting by indexed_on

snopoke committed:
Merge pull request #24754 from dimagi/sk/dd-tag-error

change submission_type tag from 'unknown' to 'error'

Rohit25negi committed:
update staging.yaml

[ci skip]

Commit:
generator does not support indexing

Commit:
__future__ imports

Commit:
use new open, pass unicode to json.loads

Rohit25negi committed:
Merge pull request #24727 from dimagi/rn_pull_dob_from_person_case

pull dob from person case for child health case

orangejenny committed:
Merge pull request #24747 from dimagi/jls/bulk-trans-mobile-ucr

Add report headers and descriptions to bulk application translations

orangejenny committed:
Updated requirements for python 3

millerdev committed:
Merge pull request #24750 from dimagi/je/remove-task

Remove missing doc task

millerdev committed:
Merge pull request #24753 from dimagi/sr-cancel

Fix the method to return a boolean

orangejenny committed:
Removed ability to data clean case name and type

orangejenny committed:
Merge pull request #24744 from dimagi/jls/flaky

Added flaky

orangejenny committed:
Added @flaky to test_change_has_user

orangejenny committed:
Update staging.yaml

orangejenny committed:
Merge pull request #24378 from dimagi/jls/build-app-ccz-app-id-step-2-of-3

Updated signature of build_application_zip

orangejenny committed:
Merge pull request #24756 from dimagi/jls/limit-data-cleaning-system-properties

Removed ability to data clean case name and type

nickpell committed:
Merge pull request #24755 from dimagi/modernize

run modernize

calellowitz committed:
Merge pull request #24656 from lwyszomi/lw/icds-644

[ICDS] month filter in SSD report start from Feb 2018

calellowitz committed:
Revert "[ICDS] month filter in SSD report start from Feb 2018"

calellowitz committed:
Merge pull request #24762 from dimagi/revert-24656-lw/icds-644

Revert "[ICDS] month filter in SSD report start from Feb 2018"

orangejenny committed:
Merge pull request #24757 from dimagi/jls/flaky-test_change_has_user

Added @flaky to test_change_has_user

Commit:
Revert "Revert "[ICDS] month filter in SSD report start from Feb 2018""

This reverts commit reverting in case of continue work on icds-644.

Commit:
[ICDS-644] Display modal after SDD open

When date is invalid prompt box with month and year selection
will be displayed after navigating to service delivery dashboard

lwyszomi commented on custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js:
```diff
@@ -6,7 +6,8 @@ function MonthModalController($location, $uibModalInstance) {
     vm.monthsCopy = [];
-
+    vm.showMessage = false;
+    var isSDD =  $location.path().indexOf('service_delivery_dashboard') != -1;
```
should be ```===```

lwyszomi commented on custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js:
```diff
@@ -108,6 +115,23 @@ function MonthFilterController($scope, $location, $uibModal, storageService) {
+    var init = function() {
+            var month = $location.search()['month'];
+            var year = $location.search()['year'];
+            var display_modal = true;
```
please use camel case in JS variables

lwyszomi commented on custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js:
```diff
@@ -27,6 +28,12 @@ function MonthModalController($location, $uibModalInstance) {
     vm.selectedMonth = $location.search()['month'] !== void(0) ? parseInt($location.search()['month']) : new Date().getMonth() + 1;
     vm.selectedYear = $location.search()['year'] !== void(0) ? parseInt($location.search()['year']) : new Date().getFullYear();
 
+    if(isSDD && (vm.selectedYear<2019||vm.selectedYear==2019&&vm.selectedMonth==1)){
```
add white space around operators

could you add brackets to the condition, will be more readable in future

lwyszomi commented on custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js:
```diff
@@ -108,6 +115,23 @@ function MonthFilterController($scope, $location, $uibModal, storageService) {
+            var year = $location.search()['year'];
+            var display_modal = true;
+
+            if(year > 2019 || year==2019&&month>1){
```
add white spaces around operators

add brackets to the condition, will be more readable in future

Commit:
Pull request review changes

Commit:
Add unit test for auto modal display in SDD

Added tests to check if modal is displayed when incorrect data is passed during navigation

Commit:
Changed function calls assertions in month-filter tests

Changed assertions from chai.expect(function).to.have.been.called
to assert(function.calledOnece)

lwyszomi commented on custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js:
```diff
@@ -131,7 +131,7 @@ function MonthFilterController($scope, $location, $uibModal, storageService) {
     }
 
-    init();
+    this.init();
```
this should be ```vm.init()```
